app/bin/tools/task_stats.dart

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2018, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// Creates a report about analysis and dartdoc task failures.
+/// Example use:
+///   dart bin/tools/task_stats.dart --output report.json
+
+import 'dart:async';
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:appengine/appengine.dart';
+import 'package:args/args.dart';
+import 'package:gcloud/db.dart';
+import 'package:pool/pool.dart';
+
+import 'package:pub_dartlang_org/frontend/models.dart';
+import 'package:pub_dartlang_org/frontend/service_utils.dart';
+import 'package:pub_dartlang_org/shared/analyzer_client.dart';
+import 'package:pub_dartlang_org/shared/analyzer_memcache.dart';
+import 'package:pub_dartlang_org/shared/dartdoc_client.dart';
+import 'package:pub_dartlang_org/shared/dartdoc_memcache.dart';
+
+Future main(List<String> args) async {
+  final ArgParser parser = new ArgParser()
+    ..addOption('max-age',
+        defaultsTo: '365', help: 'The maximum age of the package in days.')
+    ..addOption('output', help: 'The report output file (or stdout otherwise)');
+  final ArgResults argv = parser.parse(args);
+  final int age = int.parse(argv['max-age']);
+
+  final pool = new Pool(20);
+
+  final Map report = {};
+
+  await withProdServices(() async {
+    registerAnalyzerMemcache(new AnalyzerMemcache(memcacheService));
+    registerDartdocMemcache(new DartdocMemcache(memcacheService));
+    registerAnalyzerClient(new AnalyzerClient());
+    registerDartdocClient(new DartdocClient());
+
+    final statFutures = <Future<_Stat>>[];
+    final twoYearsAgo = new DateTime.now().subtract(new Duration(days: age));
```

Comment:
`ageLimit`? It's not really two years now.

Reply:
Good catch. Using `maxAgeDays` and `updatedAfter` instead.